### PR TITLE
{w,y,z}*: replace `plist` with `service` block

### PR DESCRIPTION
**Pull request by branchv**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
See #79367

### CODE REVIEW

File: Formula/zrepl.rb [hunk 1]
```diff
@@ -32,42 +32,14 @@ def post_install
...
-      </plist>
-    EOS
+  plist_options startup: true
+  service do
+    run [opt_bin/"zrepl", "daemon"]
+    keep_alive true
+    working_dir var/"run/zrepl"
+    log_path var/"log/zrepl/zrepl.out.log"
+    error_log_path var/"log/zrepl/zrepl.err.log"
+    environment_variables PATH: "/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:#{HOMEBREW_PREFIX}/bin"
```
Comment (SMillerDev) [inline Formula/zrepl.rb:42]: Didn't we have a variable for this? I think appium is using it.

Comment (branchvincent) [reply Formula/zrepl.rb]: thanks, fixed

File: Formula/yubikey-agent.rb [hunk 2]
```diff
@@ -31,38 +31,12 @@ def caveats
...
-        <string>#{var}/log/yubikey-agent.log</string>
-      </dict>
-      </plist>
-    EOS
+  service do
+    run [opt_bin/"yubikey-agent", "-l", var/"run/yubikey-agent.sock"]
+    keep_alive true
+    log_path var/"log/yubikey-agent.log"
+    error_log_path var/"log/yubikey-agent.log"
+    environment_variables PATH: "/usr/bin:/bin:/usr/sbin:/sbin"
```
Comment (SMillerDev) [inline Formula/yubikey-agent.rb:39]: AFAIK this is the default for services.

Comment (branchvincent) [reply Formula/yubikey-agent.rb:39]: removed!

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**